snapshots/store_test.go

```diff
@@ -0,0 +1,366 @@
+package snapshots_test
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	db "github.com/tendermint/tm-db"
+
+	"github.com/cosmos/cosmos-sdk/snapshots"
+	"github.com/cosmos/cosmos-sdk/snapshots/types"
+)
+
+func setupStore(t *testing.T) (*snapshots.Store, func()) {
+	tempdir, err := ioutil.TempDir("", "snapshots")
+	require.NoError(t, err)
+
+	store, err := snapshots.NewStore(db.NewMemDB(), tempdir)
+	require.NoError(t, err)
+
+	_, err = store.Save(1, 1, makeChunks([][]byte{
+		{1, 1, 0}, {1, 1, 1},
+	}))
+	require.NoError(t, err)
+	_, err = store.Save(2, 1, makeChunks([][]byte{
+		{2, 1, 0}, {2, 1, 1},
+	}))
+	require.NoError(t, err)
+	_, err = store.Save(2, 2, makeChunks([][]byte{
+		{2, 2, 0}, {2, 2, 1}, {2, 2, 2},
+	}))
+	require.NoError(t, err)
+	_, err = store.Save(3, 2, makeChunks([][]byte{
+		{3, 2, 0}, {3, 2, 1}, {3, 2, 2},
+	}))
+	require.NoError(t, err)
+
+	teardown := func() {
+		err := os.RemoveAll(tempdir)
+		if err != nil {
+			t.Logf("Failed to remove tempdir %q: %v", tempdir, err)
+		}
+	}
+	return store, teardown
+}
```

Comment:
I'm not sure table-driven tests would be beneficial for these sorts of tests, since they depend on a relatively large amount of state and the test logic can be complex. We'd basically end up with a big switch/case for each of the test cases, in which case there's no real point to it.

Reply:
Also, the code you're referring to is just a fixture, not a test. It sets up the pre-existing state for all the other tests.